google-oauth-client-jetty/src/main/java/com/google/api/client/extensions/jetty/auth/oauth2/LocalServerReceiver.java

```diff
@@ -128,22 +119,16 @@ public String getRedirectUri() throws IOException {
 
   @Override
   public String waitForCode() throws IOException {
-    lock.lock();
-    try {
-      while (code == null && error == null) {
-        gotAuthorizationResponse.awaitUninterruptibly();
-      }
-      if (error != null) {
-        throw new IOException("User authorization failed (" + error + ")");
-      }
-      return code;
-    } finally {
-      lock.unlock();
+    waitUnlessSignaled.acquireUninterruptibly();
+    if (error != null) {
```

Comment:
It looks like this won't fail throw if `stop()` is called. So `waitForCode()` will return `null`. I don't really know if that's good or bad, but it should probably be in some JavaDoc.